pypsrp/client.py

```diff
@@ -286,3 +291,11 @@ def sanitise_clixml(clixml):
             output = Serializer()._deserialize_string("".join(errors))
 
         return output
+
+
+def _expand_remote_path(runspace_pool, path):
+    expand_command = "[System.Environment]::ExpandEnvironmentVariables('%s')" % (path,)
+    powershell = PowerShell(runspace_pool)
+    powershell.add_script(expand_command)
+    powershell.invoke(on_error_message="Failed to expand path")
```

Comment:
It's good that you are reusing the RunspacePool so we don't hit the cost of creating a new one but we are still sending 2 extra messages that we don't technically need to here, creating the pipeline, then starting to receive the pipeline output. Instead we can avoid all this and just expand the path as part of single command we are already running. The way you can do this is something like the following

```python
ps = PowerShell(pool)

# Inject the raw src/dest from the client as a variable so we don't need to escape any chars
ps.add_cmdlet('Set-Variable). \
    add_parameter('Name', 'Path'). \
    add_parameter('Value', src). \
    add_statement()
    # add_statement() is a way of splitting commands from each other. Think of it as a newline or `;` in PowerShell scripts.

if expand_variables:
    # Because `$Path` was set above we need just to expand the string based on that variable and set it back to itself
    ps.add_script('$Path = [System.Environment]::ExpandEnvironmentVariables($Path)').add_statement()

# This is the command/script that we normally run in copy/fetch
ps.add_script(command)
ps.invoke()
```

Unfortunately this does mean we need to change https://github.com/jborean93/pypsrp/blob/master/pypsrp/pwsh_scripts/copy.ps1#L105 and https://github.com/jborean93/pypsrp/blob/master/pypsrp/pwsh_scripts/fetch.ps1#L10 to use `$Path` and change those scripts to rename any existing `$path` vars to something else to avoid any collisions.

Reply:
well, this is pretty cool. Took me a moment to realize what you mean but I like it! I did not realize I can extend the script like this.

I will update the PR to reflect these changes. If I am not mistaken I can also add some expansion message to the script so it's possible to see the final path after expansion somewhere. (Could be useful for debugging I guess)

Reply:
I believe the function is meant to return the absolute path that was used. I was doing that for relative paths and it makes sense for that to apply to paths with env vars.

Reply:
I am not exactly sure what you mean right now. Which function is meant to return the absolute path?

btw when I look at the solution I wonder if there is even an advantage to do it like this. It looks like there is going to be less code added when I just extend the copy/fetch.ps1 with parameter `expand_remote_variable=True/False` and add inside one `if` to handle that

I will play around with it a little bit more today to see how it looks/works

Reply:
Oh, is it possible that if I do it this way -- add the Set-Variable command at the beginning, then it will 'eat' the input file? Because I feel like it's happening to me 😄 I am not sure how to debug the PS exactly, not much experience there but nothing is copied even though the path is properly expanded

Reply:
I pushed the variant where I extended the copy and fetch files with the expansion logic